internal/deploys/serviceInstance.go

```diff
@@ -27,6 +44,30 @@ func (sic *ServiceInstanceController) ListAll() ([]ServiceInstance, error) {
 	return sic.store.listAll()
 }
 
+// CreateNew accepts the name of an environment and a service. It will perform find or create operations for both
+// and their create an association between them
+func (sic *ServiceInstanceController) CreateNew(newServiceInstance CreateServiceInstanceRequest) (ServiceInstance, error) {
+	// check if the environment already exists
+	environmentID, err := sic.environments.FindOrCreate(newServiceInstance.EnvironmentName)
+	if err != nil {
+		return ServiceInstance{}, err
+	}
+
+	// check if the service already exists
+	serviceID, doesExist := sic.services.DoesServiceExist(newServiceInstance.ServiceName)
+	if !doesExist {
+		return ServiceInstance{}, fmt.Errorf("service: %s does not exist", newServiceInstance.ServiceName)
+	}
```

Comment:
The comment says it will do a find or create operation for the service, but it seems like if the service doesn't exist it always throws an error

Reply:
oh whoops forgot to update the that block 🤦🏻. Thanks for the catch